File: web/src/app/models/response/payment.ts
import { PaymentStatus } from '../auth';

/**
 * One row of `public.subscriptions`, which is the payment history — every row
 * is a purchase with a Razorpay reference, an amount and the term it bought.
 * There is no separate payments table, and there should not be one: it would
 * duplicate all of this and start disagreeing with it.
 */
export interface Payment {
  /** Row number for the table, not the uuid — that is not worth showing. */
  id: number;
  /** The Razorpay payment id, falling back to the order id before one exists. */
  reference: string;
  paidOn: string;
  plan: string;
  periodFrom: string;
  periodTo: string;
  amount: number;
  /** ISO 4217, from the row. Defaults to INR in the schema. */
  currency: string;
  method: string;
  status: PaymentStatus;
}

File: src-tauri/src/models/response/payment.rs
use serde::Serialize;

#[derive(Debug, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct Payment {
    pub id: i64,
    pub reference: String,
    pub paid_on: String,
    pub plan: String,
    pub period_from: String,
    pub period_to: String,
    pub amount: f64,
    pub currency: String,
    pub method: String,
    pub status: String,
}
